src/category_theory/connected.lean

```diff
@@ -0,0 +1,375 @@
+/-
+Copyright (c) 2020 Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bhavik Mehta
+-/
+
+import category_theory.limits.shapes.pullbacks
+import category_theory.limits.shapes.binary_products
+import category_theory.limits.shapes.equalizers
+import category_theory.limits.preserves
+
+/-!
+# Connected category
+
+Define a connected category
```

Comment:
The module doc should have more content:

1. explain the definition
2. say that you prove equivalence with other definitions
3. explain the main result: products preserve connected limits

It's possible we want that result in a separate file, under the `category_theory/limits/` directory.